comid/measurement.go

```diff
@@ -153,6 +153,9 @@ func (o Mkey) MarshalJSON() ([]byte, error) {
 }
 
 func (o Mkey) MarshalCBOR() ([]byte, error) {
+	if err := o.Valid(); err != nil {
+		return nil, err
+	}
```

Comment:
the validation check is usually done in the `ToCBOR` method - which, if the payload is deemed "valid", will call the custom marshaler.